### ISSUE TITLE
Support STM32CubeIDE excluded files from file generation

### ISSUE THREAD

**Issue by Imrekorf** [microsoft/vscode-embedded-tools#30]
Type: Feature Request

Currently the st-project.cmake generated from an stm32cubeide .cproject does not take into account any excluded files.|
These files can be found in the .cproject under 
```xml
<sourceEntries>
<entry excluding="file1|file2|folderpath" />
</sourceEntries>
```

**Comment by benmcmorran** [comment]
Thanks for the report @Imrekorf. What version of the Embedded Tools extension are you using? Support for exclusions should have been [added in version 0.5.0](https://marketplace.visualstudio.com/items/ms-vscode.vscode-embedded-tools/changelog), which was released on November 30 (see #29). If that's not working for you, are you able to attach a minimal project that reproduces the issue? That will help us debug the problem.

**Comment by Imrekorf** [comment]
[ExcludedFileMin.zip](https://github.com/microsoft/vscode-embedded-tools/files/10313003/ExcludedFileMin.zip)
Hi I've added the minimal project that reproduces the problem for me. I've marked the test folder inside the CustomHAL folder as "exclude from build" for both Debug and Release. 
I'm using stm32cubeide 1.11.0 and the Embedded tools V0.5.0

**Comment by benmcmorran** [comment]
Thanks! I was able to track the issue down to a bug in our handling of excluded directories. A fix should be publicly available in the pre-release version of the extension in the next few days. I'll let you know when it's available.

Were you also seeing issues with exclusions for individual files (`<entry excluding="foo/bar.c" />`)? The issue I identified only applied to directories (`<entry excluding="foo" />`) and files exclusions seemed to already work correctly, but if you were seeing issues with individual file exclusions too, I can also take a look at that.

**Comment by benmcmorran** [comment]
This issue is fixed in pre-release version `0.5.221229002` of the Embedded Tools. You can switch to the pre-release version (which updates nightly) in the VS Code Marketplace.

<img width="332" alt="image" src="https://user-images.githubusercontent.com/6981284/210019479-b5a7c09b-1ed5-4fd4-a251-bae5aaab040a.png">

**Comment by benmcmorran** [comment]
This change is in version 0.6.0 of the extension, which is now available in the VS Code Marketplace.
